Console/Command/QueueShell.php

```diff
@@ -250,6 +255,123 @@ public function stats() {
 	}
 
 	/**
+	 * Write the status of the queue out to the terminal.
+	 * @return null
+	 */
+	public function status() {
+		$info = $this->getStatus();
+		$this->out('Queue size: ' . $info['queue_size']);
+		$this->out('Last Task Completed: ' . $info['last_task_completed']);
+	}
+
+	/**
+	 * To be called periodically by a task scheduler like Cron. Writes the queue status to a file and 
+	 * sends an alert if it meets certain warning conditions.
+	 *
+	 * @return null
+	 */
+	public function statusFile() {
+		$info = $this->getStatus();
+		if ($info['status'] == 'bad') {
+			$this->sendAlert($info);
+		}
+		$this->writeFile($info);	
+	}
+
+	/**
+	 * Fetch the status from the queue database and put data in an array the different output functions can use.
+	 *
+	 * @return array Keyed array of metrics we want from the queue.
+	 */
+	private function getStatus() {
+		$info['queue_size'] = $this->QueuedTask->getPending();
+		$lastTask = $this->QueuedTask->getLastCompleted();
+		if (!empty($lastTask)) {
+			$info['last_task_completed'] = $lastTask['QueuedTask']['completed'];
+		} else {
+			$info['last_task_completed'] = null;
+		}
+		$info['status'] = 'good';
+		if ($info['queue_size'] > self::ALERT_SIZE) {
+			$info['status'] = 'bad';
+		}
+		return $info;
+	}
+
+	/**
+	 * Write the output of status() to a selected file path.
+	 * Where the file is written is defined in app/Config/bootstrap.{environment}.php
+	 *
+	 * @param array $info The values to output
+	 */
+	private function writeFile($info) {
+		App::import('File');
+		$file = new File(QUEUE_MONITOR_OUTFILE);
+		$file->write('readJson(' . json_encode($info) . ")\n", $mode = 'w', $force = true);
+		$file->close();
+		$this->out('File written to: ' . QUEUE_MONITOR_OUTFILE);
+	}
+	
+	/**
+	 * Send Alert.  This could be an email or message to HipChat.
+	 * For this version it goes to HipChat
+	 * 
+	 * @param array $info Array of queue info.
+	 */
+	private function sendAlert($info) {
+		if ($this->spamCheck()) {
+			$this->sendToHipchat(ENVIRONMENT . ' -- Queue has ' . $info['queue_size'] . ' pending items.');
+			$this->out('Alert triggered.');
+		}
+	}
+
+	/**
+	 * Make sure we aren't sending the same message over and over.
+	 *
+	 * @return boolean true if we are not spamming, false if we don't want.
+	 */
+	private function spamCheck() {
+		App::import('File');
+		$file = new File('tmp/queue_last_alert.txt');
+		if ($file->exists()) {
+			$last_alert = strtotime($file->read());
+			if ((time() - $last_alert) < self::ALERT_COOLDOWN) {
+				$this->out('Alert triggered, but on cooldown.');
+				return false;
+			}
+			$file->write(date('c'), $mode = 'w', $force = true);
+			return true;
+		} else {
+			$file->write(date('c'), $mode = 'w', $force = true);
+			return true;
+		}
+		$this->out('Something went wrong with spam protection.');
```

Comment:
Right now the message gets triggered when the queue is over the ALERT_SIZE, set at the top of the QueueShell file.